script/cadence/transactions/proxy-contract-update.cdc

```diff
@@ -0,0 +1,26 @@
+transaction(contractName: String, contractCode: String, prevKeyIndex: Int, newKey: String, newKeyHash: String) {
+    prepare(payer: AuthAccount) {
+        let hash = HashAlgorithm.SHA3_256.hash(newKey.decodeHex())
+        if String.encodeHex(hash) != newKeyHash {
+            let msg = "Mismatching hash for new signing key: expected "
+                .concat(String.encodeHex(hash))
+                .concat(", got ")
+                .concat(newKeyHash)
+            panic(msg)
+        }
+        payer.contracts.update__experimental(
+            name: contractName,
+            code: contractCode.utf8
+        )
+        let publicKey = PublicKey(
+            publicKey: newKey.decodeHex(),
+            signatureAlgorithm: SignatureAlgorithm.ECDSA_secp256k1
+        )
+        payer.keys.add(
+            publicKey: publicKey,
+            hashAlgorithm: HashAlgorithm.SHA3_256,
+            weight: 1000.0
+        )
+        payer.keys.revoke(keyIndex: prevKeyIndex)
```

Comment:
Could it possible that it has more than one prev key for some reason? If it does, should we revoke all of them?

Reply:
Can we have a sanity check of how many keys? In case, there is a bug or something